authlib/integrations/starlette_client/apps.py

```diff
@@ -97,3 +99,35 @@ async def authorize_access_token(self, request, **kwargs):
             )
             token["userinfo"] = userinfo
         return token
+
+    async def decode_access_token(self, token, alg_values=None, **kwargs):
+        claims_options = kwargs.pop("claims_options", None)
+        if claims_options is None:
+            metadata = await self.load_server_metadata()
+            # RFC9068 Section 2.2 and 4
+            claims_options = {
+                "iss": {"essential": True, "value": metadata["issuer"]},
+                "exp": {"essential": True},
+                "aud": {"essential": True, "value": self.client_id},
+                # As far as I know, there's no way to verify what the value of sub should be without being
+                # responsible for requesting the access token to begin with
+                "sub": {"essential": True},
+                "client_id": {"essential": True},
+                "iat": {"essential": True},
+                "jti": {"essential": True},
+            }
+
+        # RFC9068: authorization servers and resource servers must support RS256
+        if alg_values is None:
+            alg_values = ["RS256"]
```

Comment:
The `alg` value should be in the JWT header, so I am not sure this is needed to guess here.

You should be able to achieve the same thing with:
```
from authlib.jose import jwt
jwt.decode(token, jwks)
```

Reply:
I apologize if my understanding of JWT and Access Tokens is a little sparse, but is there no concern with accepting any algorithm in the header? I did some looking around and [this article](https://curity.io/resources/learn/jwt-best-practices/) suggests that you shouldn't necessarily trust the header, but unsure how accurate it is:

> When verifying or decrypting the token you should always check the value of this claim with a list of algorithms that your system accepts. This mitigates an attack vector where someone would tamper with the token and make you use a different, probably less secure algorithm to verify the signature or decrypt the token.

I'm fine with removing the line and assuming the header `alg` is safe to use however if that's what we want to do.